### PR TITLE
Shady nagy/api endpoints

### PR DESCRIPTION
**Pull request by ShadyNagy**
Applying Ardalis endpoints on the uploader and API authorization attribute `UploaderApiAuthorization` added.
- AddVideoInfo
- DeleteAllVideosNoVimeo
- DeleteVideo
- UpdateAllVideosThumbnails
- UpdateVideoThumbnails
- UpdateVideoThumbnailsByArchiveVideo

### CODE REVIEW

File: src/DevBetterWeb.Web/Endpoints/UpdateVideoThumbnailsByArchiveVideo.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+	[HttpPost("videos/update-video-thumbnails")]
+	public override async Task<ActionResult<ArchiveVideoDto>> HandleAsync([FromBody] ArchiveVideoDto archiveVideoDto, CancellationToken cancellationToken = default)
+	{
+		var archiveVideo = _mapper.Map<ArchiveVideo>(archiveVideoDto);
+
+		var spec = new ArchiveVideoByVideoIdSpec(archiveVideo.VideoId!);
+		var existVideo = await _repository.FirstOrDefaultAsync(spec, cancellationToken);
+		if (existVideo == null)
+		{
+			return BadRequest();
```
Comment (ardalis) [inline src/DevBetterWeb.Web/Endpoints/UpdateVideoThumbnailsByArchiveVideo.cs:37]: Why not 404?

File: src/DevBetterWeb.Web/Endpoints/UpdateVideoThumbnails.cs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+	}
+
+	[HttpGet("videos/update-video-thumbnails/{videoId}")]
+	public override async Task<ActionResult<ArchiveVideoDto>> HandleAsync(long videoId, CancellationToken cancellationToken = default)
+	{
+		var spec = new ArchiveVideoByVideoIdSpec(videoId.ToString());
+		var existVideo = await _repository.FirstOrDefaultAsync(spec, cancellationToken);
+		if (existVideo == null)
+		{
+			return BadRequest();
```
Comment (ardalis) [inline src/DevBetterWeb.Web/Endpoints/UpdateVideoThumbnails.cs:45]: NotFound()

File: src/DevBetterWeb.Web/Endpoints/UpdateVideoThumbnails.cs [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+		}
+
+		var response = await _getVideoService.ExecuteAsync(videoId.ToString(), cancellationToken);
+		if (response?.Data == null)
+		{
+			return BadRequest("Video Not Found!");
```
Comment (ardalis) [inline src/DevBetterWeb.Web/Endpoints/UpdateVideoThumbnails.cs:51]: NotFound()

File: src/DevBetterWeb.Web/Endpoints/DeleteVideo.cs [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+		_repository = repository;
+		_deleteVideoService = deleteVideoService;
+	}
+
+	[HttpDelete("videos/uploader/delete-video/{vimeoVideoId}")]
+	public override async Task<ActionResult> HandleAsync([FromRoute] string vimeoVideoId, CancellationToken cancellationToken = default)
+	{
+		var spec = new ArchiveVideoByVideoIdSpec(vimeoVideoId);
+		var existVideo = await _repository.FirstOrDefaultAsync(spec, cancellationToken);
+		if (existVideo != null)
```
Comment (ardalis) [inline src/DevBetterWeb.Web/Endpoints/DeleteVideo.cs:32]: if it is null, return NotFound()

**Comment by ShadyNagy** [comment]
> Add NotFound() then good to go!

done

**Pull request merged**